doc/specs/#885 - winrt Terminal Settings.md

```diff
@@ -0,0 +1,211 @@
+---
+author: Carlos Zamora @carlos-zamora
+created on: 2020-07-10
+last updated: 2020-07-10
+issue id: [#885](https://github.com/microsoft/terminal/issues/885)
+---
+
+# Spec Title
+
+## Abstract
+
+This spec proposes a major refactor and repurposing of the TerminalSettings project. TerminalSettings would
+ be responsible for exposing, serializing, and deserializing settings as WinRT objects for Windows Terminal. In
+ doing so, Terminal's settings model is accessible as WinRT objects to existing components like TerminalApp,
+ TerminalControl, and TerminalCore. Additionally, Terminal Settings can be used by the Settings UI or
+ Shell Extensions to modify or reference Terminal's settings respectively.
+
+## Inspiration
+
+The main driver for this change is the Settings UI. The Settings UI will need to read and modify Terminal Settings.
+ At the time of writing this spec, the Terminal Settings are serialized as objects in the TerminalApp project.
+ To access these objects, the Settings UI would need to be a part of TerminalApp too, making it more bloated.
+
+## Solution Design
+
+### Terminal Settings Model: Objects and Projections
+
+The objects/interfaces introduced in this section will be exposed as WinRT objects/interfaces
+ within the `Microsoft.Terminal.Settings.Model` namespace. This allows them to be interacted with across
+ different project layers.
+
+Additionally, the top-level object holding all of the settings related to Windows Terminal will be
+`AppSettings`. The runtime class will look something like this:
+```c++
+runtimeclass AppSettings
+{
+    // global settings that generally only affect TerminalApp
+    GlobalSettings Globals { get; };
+
+    // the list of profiles available to Windows Terminal (custom and dynamically generated)
+    IObservableVector<Profile> Profiles { get; };
+
+    // the list of key bindings indexed by their unique key chord
+    IObservableMap<KeyChord, Action> Keybindings { get; };
+
+    // the list of command palette commands indexed by their name (custom or auto-generated)
+    IObservableMap<String, Action> Commands { get; };
+
+    // the list of color schemes indexed by their unique names
+    IObservableMap<String, ColorScheme> Schemes { get; };
+
+    // a list of warnings encountered during the serialization of the JSON
+    IVector<SerializationWarnings> Warnings { get; };

```

Comment:
No. That would be the responsibility of the TermApp with the `TerminalSettings` bridge. Since TermApp just has a reference to the `AppSettings`, TermApp would use that to construct a `TerminalSettings`, then hand over to the TermControl.